web/src/components/pair/PairCodeMatchEditor.vue

```diff
@@ -0,0 +1,349 @@
+<template>
+  <div class="editor">
+    <div class="editor-title text--primary">
+      <v-icon left>mdi-file-outline</v-icon>
+      {{ file.shortPath }}
+    </div>
+
+    <div ref="editorElem" class="editor-element"></div>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import {
+  withDefaults,
+  ref,
+  shallowRef,
+  onMounted,
+  onUnmounted,
+  watch,
+  computed,
+} from "vue";
+import { Pair, Metadata, Match } from "@/api/models";
+import * as monaco from "monaco-editor";
+import { useVModel } from "@vueuse/core";
+
+interface Props {
+  side: "left" | "right";
+  pair: Pair;
+  metadata: Metadata;
+  selectedMatch: Match | null;
+  hoveringMatch: Match | null;
+}
+
+interface Selection {
+  match: Match | null;
+  range: monaco.IRange | null;
+  isWholeLine: boolean;
+}
+
+const props = withDefaults(defineProps<Props>(), {});
+const emit = defineEmits(["update:selectedMatch"]);
+
+// Colors for the different selections.
+const colors = {
+  match: "rgba(60, 115, 168, 0.2)",
+  matchHovering: "rgba(60, 115, 168, 0.3)",
+  matchSelected: "rgba(26, 188, 156, 0.3)",
+};
+
+// File to display
+// Based on the pair & the given side.
+const file = computed(() => props.side === "left" ? props.pair.leftFile : props.pair.rightFile);
+
+// List of matches, sorted on the start line & column.
+const matches = computed(() => {
+  const matches = [...props.pair.matches];
+  if (!matches) return [];
+
+  return matches.sort(
+    (a, b) =>
+      (a[props.side].startRow - b[props.side].startRow) * 10000 +
+      (a[props.side].startCol - b[props.side].startCol)
+  );
+});
+// Selected match
+const selectedMatch = useVModel(props, "selectedMatch", emit);
+// Hovering match
+const hoveringMatch = useVModel(props, "hoveringMatch", emit);
+
+// Editor template ref
+const editorElem = ref();
+// Monaco editor
+const editor = shallowRef<monaco.editor.IEditorOverrideServices>();
+// List of selections
+const selections = ref<Selection[]>([]);
+// Monaco editor decorations
+const decorations = shallowRef([]);
+
+// Get the match at a given editor position.
+const getMatchAtPosition = (row: number, col: number): Match | null => {

```

Comment:
It seems like some of the highlighted matches are "behind" others, making it not possible to select them. I think this may be fixed by trying to find the smallest match on the current position?